### ISSUE TITLE
route-maker JS Error on IE6 and IE7

### ISSUE THREAD

**Issue by aljimenez** [YahooArchive/mojito#1121]
The array returned by String.match is being iterated through a forin and, as a result, non-string properties such as index are evaluated and substr fails with the error below. Check the [match method on IE](http://msdn.microsoft.com/en-us/library/7df7sf4x.aspx).

Line 121: Object doesn't support property or method 'substr'

``` javascript
matches = route.path.match(/:([^\/]+)/g);
for (i in matches) {
    if (matches.hasOwnProperty(i)) {
        route.requires[matches[i].substr(1)] = '[^&]+';
    }
}
```

**Comment by isao** [comment]
fixed by #1152

**Comment by isao** [comment]
fyi fix included in mojito@0.6.1 which was just published earlier today.

**Comment by isao** [comment]
@aljimenez sorry I didn't realize you reported this too, so much earlier. Thanks for the report and apologies for delay!
